### PR TITLE
Throw on unsupported

### PR DESCRIPTION
**Pull request by mrsimpson**
This shall enhance the transpiler so that y consumer can request more strict checks

### CODE REVIEW

File: packages/transpiler/src/index.ts [hunk 1]
```diff
@@ -88,8 +102,11 @@ export class Transpiler {
         return transpiler.transpile(node);
       }
     }
-
-    return "todo, statement: " + node.get().constructor.name;
+    if (this.options.checks === CHECKS.comment) {
+      return "todo, statement: " + node.get().constructor.name;
+    } else {
+      throw new Error(`Statement ${node.get().constructor.name} not supported`);
```
Comment (mrsimpson) [inline packages/transpiler/src/index.ts]: @larshp I added options  to the transpiler to react on that as per the wishes of the consumer

**Comment by mrsimpson** [comment]
@larshp please review

File: packages/transpiler/src/index.ts (hunk 1 above)
Comment (larshp) [reply packages/transpiler/src/index.ts]: think its okay to only just throw the error

File: test/running_examples.ts [hunk 2]
```diff
@@ -167,4 +167,31 @@ describe("Full Examples", () => {
...
   });
 
+  it("Should throw an error if invalid code is requested to be transpiled", () => {
+    const code = `THIS IS NOT ABAP.`;
+    try {
+      new Transpiler().run(code);
+      throw new Error("An exception should have been raised for invalid code");
+    } catch (e) {
+      // expected - do nothing
+    }
```
Comment (larshp) [inline test/running_examples.ts]: in chai its possible to do something like

`expect(() => {something;}).to.throw();`

**Comment by mrsimpson** [comment]
@larshp once this is merged, can you please publish once more?

File: packages/transpiler/src/index.ts (hunk 1 above)
Comment (larshp) [reply packages/transpiler/src/index.ts]: I think its okay not to have any options, and just throw the exception here

Comment (mrsimpson) [reply packages/transpiler/src/index.ts]: I can imagine having options (particularly checks) as valuable in future. Does it hurt, shall I remove it?

Comment (larshp) [reply packages/transpiler/src/index.ts]: yes, keep the throw, remove the options

there will be options, but its always a mess to have multiple behaviors

**Comment by mrsimpson** [comment]
True. As ABAPer used to customizing, I am a frequent [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) victim

**Comment by mrsimpson** [comment]
Should we adapt the frontend to propagate exceptions?

**Comment by larshp** [comment]
web? yea, just catch the exception and write it to `editor3`

**Comment by larshp** [comment]
transpiler 0.2.2 published

**Pull request merged**